### PR TITLE
fix lightingQueue memory leak for v3.1

### PR DESCRIPTION
**Pull request by troublemaker52025**

### CODE REVIEW

File: cocos/renderer/pipeline/RenderAdditiveLightQueue.cpp [hunk 1]
```diff
@@ -469,5 +450,4 @@ gfx::DescriptorSet *RenderAdditiveLightQueue::getOrCreateDescriptorSet(const Lig
     return _descriptorSetMap.at(light);
 }
 
-} // namespace pipeline
-} // namespace cc
+} // namespace cc::pipeline
```
Comment (YunHsiao) [inline cocos/renderer/pipeline/RenderAdditiveLightQueue.cpp:453]: Nested namespace rule is disabled in the config, if you changed this due to linter errors, you might need to double-check your `clang-tidy` environment setup.

Comment (troublemaker52025) [reply cocos/renderer/pipeline/RenderAdditiveLightQueue.cpp:453]: sure

**Review by caryliu1999 — APPROVED** [review]

File: cocos/renderer/pipeline/RenderAdditiveLightQueue.cpp [hunk 2]
```diff
@@ -47,7 +47,7 @@
 namespace cc {
 namespace pipeline {
 
-RenderAdditiveLightQueue::RenderAdditiveLightQueue(RenderPipeline *pipeline) : _pipeline(static_cast<ForwardPipeline *>(pipeline)),
+RenderAdditiveLightQueue::RenderAdditiveLightQueue(RenderPipeline *pipeline) : _pipeline(dynamic_cast<ForwardPipeline *>(pipeline)),
```
Comment (minggo) [inline cocos/renderer/pipeline/RenderAdditiveLightQueue.cpp:50]: why use dynamic_cast?

Comment (troublemaker52025) [reply cocos/renderer/pipeline/RenderAdditiveLightQueue.cpp:50]: When performing downstream conversions (base class rotor subClass), dynamic_cast has type checking and is safer than static_cast.

Comment (troublemaker52025) [reply cocos/renderer/pipeline/RenderAdditiveLightQueue.cpp:50]: This is what VisualStudio2019 recommends

**Pull request merged**